text/0000-otlp-over-messaging-systems.md

```diff
@@ -0,0 +1,110 @@
+# OTLP over messaging systems
+
+Best practices for transporting OTLP data over a message system.
+
+## Motivation
+
+This proposal tries to bring consistency and guidelines when transporting
+OTLP data over messaging systems. A non-exclusive list of examples
+of products in this category are:
+
+* Apache Kafka
+* Apache Pulsar
+* Google Pubsub
+* AWS SNS
+* RabbitMQ
+
+Using an intermediate messaging system is in some cases preferred compared to
+a direct network connection between the telemetry producer and consumer.
+Reasons for using such an intermediate medium could be:
+
+* Protecting against backend failure
+* Security Policies
+* Network Policies
+* Buffering
+
+An extra motivation to have a consistent definition of the payload is that
+it would be easy to transfer the OTLP data from one messaging system to
+another just by reading the payload from one system and writing it to
+another without the need for transformations.
+
+## Explanation
+
+Because the OTLP payload that is sent over messaging systems is
+well-defined, it’s easy to implement new systems consistently.
+An implementation should at least support `otlp_proto`, meaning that the
+payload is Protobuf serialized from `ExportTraceServiceRequest` for traces,
+`ExportMetricsServiceRequest` for metrics, or `ExportLogsServiceRequest` for
+logs.
+
+Optionally an implementation could support `otlp_json` or other alternative
+encodings like jaeger or zipkin payloads. Alternative encodings should be configured
+by an `encoding` field in the configuration file.
+
+For the user it’s beneficial that both an exporter, and a receiver are
+implemented in the OpenTelemetry Collector, but SDK developers could
+also implement a language specific exporter.
+
+For log data, implementors of a receiver are encouraged to also support
+receiving raw data from a topic. This enables scenarios were
+non-OpenTelemetry log producers can produce a stream of logs, either
+structured or unstructured, on that topic and the receiver wraps the log
+data in a configurable resource.
```

Comment:
This seems unrelated to OTLP. Should it be in this OTEP?

Reply:
Maybe not, it would be nice though that implements do this. But maybe it shouldn't be in the spec.